### PR TITLE
add support for quay.io

### PR DESCRIPTION
**Pull request by chennqqi**
Signed-off-by: chennqqi <chennqqi@qq.com>

### CODE REVIEW

File: docker_wrapper.py [hunk 1]
```diff
@@ -4,10 +4,33 @@
 import os
 import sys
 
-mirror = "gcr.azk8s.cn"
-namespace = "google_containers"
-prefix = "gcr.io"
-specialPrefix = "k8s.gcr.io"
+#  xxx:yyy -> dockerhub.azk8s.cn/library/xxx:yyy
+#
```
Comment (silenceshell) [inline docker_wrapper.py]: please remove the empty line begin comments

File: docker_wrapper.py [hunk 2]
```diff
@@ -4,10 +4,33 @@
...
+#  gcr.io/xxx/yyy:zzz -> gcr.azk8s.cn/xxx/yyy:zzz
+#			
+#  k8s.gcr.io/xxx:yyy => gcr.io/google-containers/xxx:yyy -> gcr.azk8s.cn/google-containers/xxx:yyy
+#		
+#	 quay.io/xxx/yyy:zzz -> quay.azk8s.cn/xxx/yyy:zzz
```
Comment (silenceshell) [inline docker_wrapper.py]: please remove the tab ahead of this line

File: docker_wrapper.py [hunk 3]
```diff
@@ -4,10 +4,33 @@
...
+#
+
+converts = [
+	{
```
Comment (silenceshell) [inline docker_wrapper.py]: please use four blanks instead of tab

File: docker_wrapper.py [hunk 4]
```diff
@@ -4,10 +4,33 @@
...
+			'prefix': 'gcr.io',
+			'replace': lambda x: x.replace('gcr.io', 'gcr.azk8s.cn'),
+	},
+	{
+			'prefix': 'k8s.gcr.io',
+			'replace': lambda x: x.replace('k8s.gcr.io', 'gcr.azk8s.cn/google-containers'),
+	},
+	{
+			'prefix': 'quay.io',
+			'replace': lambda x: x.replace('gcr.io', 'gcr.azk8s.cn'),
```
Comment (silenceshell) [inline docker_wrapper.py]: should be `quay.io` and `quay.azk8s.cn`

File: docker_wrapper.py [hunk 5]
```diff
@@ -23,31 +46,28 @@ def usage():
         usage()
         sys.exit(-1)
 
-    # image name like k8s.gcr.io/kube-apiserver:v1.14.1 or gcr.io/google_containers/kube-apiserver:v1.14.1
```
Comment (silenceshell) [inline docker_wrapper.py:26]: do not remove this comment, it is useful for quick reminding.

File: docker_wrapper.py [hunk 6]
```diff
@@ -4,10 +4,33 @@
...
+#			
+#  k8s.gcr.io/xxx:yyy => gcr.io/google-containers/xxx:yyy -> gcr.azk8s.cn/google-containers/xxx:yyy
+#		
+#	 quay.io/xxx/yyy:zzz -> quay.azk8s.cn/xxx/yyy:zzz
+#
+
+converts = [
+	{
+			'prefix': '',
+			'replace': lambda x : 'dockerhub.azk8s.cn/library/' + x,
```
Comment (silenceshell) [inline docker_wrapper.py]: user may have set docker mirror and do not want to use own Azure mirror, so we should pull image directly.
I think this is not necessary.

**Review by silenceshell — CHANGES_REQUESTED** [review]
@chennqqi Thanks for your PR!
here are some review comments before merging:

**Comment by chennqqi** [comment]
plz review again

**Comment by silenceshell** [comment]
Ah...Please do not use [dos] style.

ref [here](https://help.github.com/cn/articles/configuring-git-to-handle-line-endings#global-settings-for-line-endings)

**Comment by chennqqi** [comment]
to unix style

**Comment by chennqqi** [comment]
change to unix style

**Comment by silenceshell** [comment]
hi @chennqqi 

I have tested your code in my linux machine, and find out that the docker_wrapper.py is changed to `UTF-8 Unicode (with BOM) text executable`, and this will cause an error when exec:

```
./docker_wrapper.py pull gcr.io/google_containers/kube-apiserver:v1.14.1
./docker_wrapper.py: line 1: #!/usr/bin/python: No such file or directory
./docker_wrapper.py: line 4: import: command not found
./docker_wrapper.py: line 5: import: command not found
./docker_wrapper.py: line 11: converts: command not found
./docker_wrapper.py: line 14: syntax error near unexpected token `('
./docker_wrapper.py: line 14: `        'replace': lambda x: x.replace('gcr.io', 'gcr.azk8s.cn'),'
```

Please make sure the script could run on linux machine, for it is mainly used in this environment.

**Comment by chennqqi** [comment]
convert dos to unix sytle shourd remove UTF-8 BOM

File: docker_wrapper.py [hunk 7]
```diff
@@ -23,31 +38,29 @@ def usage():
...
     image = sys.argv[2]
     imageArray = image.split("/")
-    
-    if imageArray[0] != prefix and imageArray[0] != specialPrefix:
-        cmd = "docker pull {image}".format(image=image)
+   
+    newImage = '' 
+    for cvt in converts:
+        if imageArray[0] == cvt['prefix']:
+    	      # image name like k8s.gcr.io/kube-apiserver:v1.14.1 or gcr.io/google_containers/kube-apiserver:v1.14.1
```
Comment (silenceshell) [inline docker_wrapper.py:47]: please move this comment before line 8.

**Review by silenceshell — CHANGES_REQUESTED** [review]
hi please move this comment before line 8.

And squash your commit into one single commit. [Ref](https://github.com/todotxt/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit)

**Comment by silenceshell** [comment]
@chennqqi thanks for your PR! 👍

**Pull request merged**